nativelib/src/main/scala/scala/scalanative/unsafe/CVarArgList.scala

```diff
@@ -75,6 +80,58 @@ object CVarArgList {
         words
     }
 
+  private def toCVarArgList_Arm64_MacOS(
+      varargs: Seq[CVarArg]
+  )(implicit z: Zone): CVarArgList = {
+    var storage = new Array[Long](registerSaveWords)
+    var wordsUsed = storage.size
+    var gpRegistersUsed = 0
+    var fpRegistersUsed = 0
+
+    def appendWord(word: Long): Unit = {
+      if (wordsUsed == storage.size) {
+        val newstorage = new Array[Long](storage.size * 2)
+        System.arraycopy(storage, 0, newstorage, 0, storage.size)
+        storage = newstorage
+      }
+      storage(wordsUsed) = word
+      wordsUsed += 1
+    }
+
+    varargs.foreach { vararg =>
+      val encoded = encode(vararg.value)(vararg.tag)
+      val isDouble = isPassedAsDouble(vararg)
+
+      if (isDouble && fpRegistersUsed < countFPRegisters) {
+        var startIndex =
+          countGPRegisters + (fpRegistersUsed * fpRegisterWords)
+        encoded.foreach { w =>
+          storage(startIndex) = w
+          startIndex += 1
+        }
+        fpRegistersUsed += 1
+      } else if (encoded.size == 1 && !isDouble && gpRegistersUsed < countGPRegisters) {
+        val startIndex = gpRegistersUsed
+        storage(startIndex) = encoded(0)
+        gpRegistersUsed += 1
+      } else {
+        encoded.foreach(appendWord)
+      }
+    }
+    val resultStorage =
+      z.alloc(sizeof[Long] * storage.size.toULong).asInstanceOf[Ptr[Long]]
+    val storageStart = storage.asInstanceOf[LongArray].at(0)
+    libc.memcpy(
+      toRawPtr(resultStorage),
+      toRawPtr(storageStart),
+      wordsUsed.toULong * sizeof[Long]
+    )
+
+    // On MacOS Arm64 the variadic arguments list is stored in memory
+    // without a header structure (such as one used on x86)
+    new CVarArgList(toRawPtr(storageStart))
```

Comment:
Since the only change is the absence of the header, let's keep it simple with only a single implementation for now, in case if in the future we would locate more inconsistencies we might refactor it to make addressing them easier. For now, I believe we should just rename previous method to `toCVarArgList_Unix` and use the runtime condition to check if the header is used.